homework/index.js

```diff
@@ -28,27 +28,74 @@
     });
     return elem;
   }
+  function NewDate(dateString) {
+    const dateTime = new Date(dateString);
+    return dateTime.toLocaleString();
+  }
+
+  function renderRepoDetails(repo, section) {
+    const Mix = createAndAppend('ul', section);
+    const name = createAndAppend('li', Mix, {
+      text: 'Repository:',
+      class: 'left-side',
+    });
+    const description = createAndAppend('li', Mix, {
+      text: 'Description:',
+      class: 'left-side',
+    });
+    const fork = createAndAppend('li', Mix, {
+      text: 'Forks:',
+      class: 'left-side',
+    });
+    const time = createAndAppend('li', Mix, {
+      text: 'Updated:',
+      class: 'left-side',
+    });
 
-  function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+    // now I'm going to use that name,description,fork,time in the below function
+
+    createAndAppend('a', name, {
+      href: repo.clone_url,
+      target: '_blank',
+      class: 'right-side-1',
+      text: repo.name,
+    });
+    createAndAppend('p', description, {
+      text: repo.description,
+      class: 'right-side-2',
+    });
+    createAndAppend('p', fork, { text: repo.forks, class: 'right-side-3' });
+    createAndAppend('p', time, {
+      text: NewDate(repo.updated_at),
+      class: 'right-side-4',
+    });
+  }
+
+  function sortName(first, second) {
+    return first.name.localeCompare(second.name);
   }
 
   function main(url) {
     fetchJSON(url, (err, repos) => {
       const root = document.getElementById('root');
+      createAndAppend('h1', root, {
+        class: 'Header',
+        text: 'HYF-Repositories',
+      });
+
       if (err) {
         createAndAppend('div', root, {
           text: err.message,
           class: 'alert-error',
         });
         return;
       }
-      const ul = createAndAppend('ul', root);
-      repos.forEach(repo => renderRepoDetails(repo, ul));
+      const section = createAndAppend('section', root);
+
+      repos.sort(sortName).forEach(repo => renderRepoDetails(repo, section));
```

Comment:
Since the function body of `sortName` is only a single line, I would prefer to code this function in-line as an arrow function:

```js
repos
  .sort((a, b) => a.name.localeCompare(b.name))
  .forEach(repo => renderRepoDetails(repo, repoContainer));
```

Reply:
thank you for this information because this is more short way